### PR TITLE
Add Any23 dataformat example.

### PR DESCRIPTION
**Pull request by betoflowv93**
This commit adds an Any23 dataformat example.

Your feedback would be appreciated :)

### CODE REVIEW

File: examples/camel-example-any23/pom.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-any23</artifactId>
+        </dependency>
+        
+        <dependency>
+        <groupId>org.apache.camel</groupId>
+        <artifactId>camel-jsonpath</artifactId>
+        </dependency>
```
Comment (bedlaj) [inline examples/camel-example-any23/pom.xml]: I dont see jsonpath used, is this dependency needed?

File: examples/camel-example-any23/pom.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    <artifactId>camel-example-any23</artifactId>
+    <name>Camel :: Example :: Any23</name>
+    <description>An example that uses the Any23 dataformat</description>
+    <packaging>jar</packaging>
+    
+    
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <camel.version>3.0.0-SNAPSHOT</camel.version>
```
Comment (bedlaj) [inline examples/camel-example-any23/pom.xml]: This property is unused

File: examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Any23StartupMessage.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ */
+package org.apache.camel.example.any23;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.StartupListener;
+
+public class Any23StartupMessage implements StartupListener {
+
+  public Any23StartupMessage() {
```
Comment (bedlaj) [inline examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Any23StartupMessage.java]: This constructor is empty and thus can be removed.

File: examples/camel-example-any23/pom.xml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (bedlaj) [inline examples/camel-example-any23/pom.xml]: Please add license header

File: examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Any23RouteBuilder.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    from("direct:extractMoreData")
+            .log("Split ${body}")
+            .convertBodyTo(String.class)
+            .toD("${body}")
+            .unmarshal()
+            //Extract RDF data of the leaders as JSONLD
+            .any23(BASEURI, Any23Type.JSONLD)
+            .log(" Result : ${body} ")
+            .to("mock:result");
```
Comment (bedlaj) [inline examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Any23RouteBuilder.java]: mock endpoint is useless here. `log:result` may be better.

File: examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Any23RouteBuilder.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    from("direct:start")
+            .log("Querying dbpedia:Ecuador ")
+            .to("http://dbpedia.org/page/Ecuador")
+            .unmarshal()
+            .any23(BASEURI)
+            .process(new Processor() {
+              public void process(Exchange exchange) throws Exception {
+                ValueFactory vf = SimpleValueFactory.getInstance();
+                Model model = (Model) exchange.getIn().getBody();
```
Comment (bedlaj) [inline examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Any23RouteBuilder.java]: Better use getBody(Class) instead this cast
```suggestion
                Model model = exchange.getIn().getBody(Model.class);
```

File: examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Any23RouteBuilder.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            })
+            .log(" Content: ${body} ")
+            //Process each leader in a separate route.
+            //In order to extract more information.
+            .split(simple("${body}"))
+            .to("direct:extractMoreData");
+
+    from("direct:extractMoreData")
+            .log("Split ${body}")
+            .convertBodyTo(String.class)
```
Comment (bedlaj) [inline examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Any23RouteBuilder.java]: If I am looking correctly, the body is already String. Why this conversion?

File: examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Application.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ */
+package org.apache.camel.example.any23;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.impl.DefaultCamelContext;
+
+public class Application {
+
+  public static void main(String[] args) throws Exception {
+    System.out.println("Camel is started. Ready to run Any23 example!");
```
Comment (bedlaj) [inline examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Application.java]: It is a bit nitpicking, but Camel is not started yet at the time of this message :-)

File: examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Application.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+    CamelContext context = new DefaultCamelContext();
+    context.start();
```
Comment (bedlaj) [inline examples/camel-example-any23/src/main/java/org/apache/camel/example/any23/Application.java]: You should add routes and startup listener before calling `context.start()`.

**Comment by oscerd** [comment]
Thanks, this has been merged.

**Pull request closed without merging**